### ISSUE TITLE
Question: HTTP OPTIONS support

### ISSUE THREAD

**Issue by Littermonkey** [openfaas/faas#774]
## Steps to Reproduce (for bugs)

1.  
2.
3.
4.

## Context

I have done  a API, Option is displayed when I request with ajax. so can you tell me how to create method with option. 
## Your Environment

* Docker version `docker version` (e.g. Docker 17.0.05 ):

* Are you using Docker Swarm or Kubernetes (FaaS-netes)?
docker swarm
* Operating System and version (e.g. Linux, Windows, MacOS):
Ubuntu 16.04.4 LTS
* Link to your project or a code example to reproduce issue:

* Please also follow the [troubleshooting guide](https://github.com/openfaas/faas/blob/master/guide/troubleshooting.md) and paste in any other diagnostic information you have:

**Comment by ivanayov** [comment]
Hi,

Can you please share more details about your function and context?

Thanks

**Comment by ivanayov** [comment]
Derek add label: support

**Comment by ivanayov** [comment]
Derek add label: question

**Comment by Littermonkey** [comment]
hi,
    Is there any 'option' get' put' other than 'POST' in our API's HTTP methods? when I use ajax to call the api it send twice requests , One is 'option'(Preflighted Requests) and the other is 'post'.

**Comment by alexellis** [comment]
Supported methods are currently GET and POST. We could potentially implement `OPTIONS` in the API gateway to return those values?

Alex

**Comment by Littermonkey** [comment]
OK  ,  what can i config for supported GET methods ?

**Comment by alexellis** [comment]
Both `GET` and `POST` are supported verbs for all functions. It's up to you to handle them inside your function code.

**Comment by alexellis** [comment]
d

**Comment by lmoran** [comment]
If I may chime in, OPTIONS method is necessary for CORS, and HEAD is useful to invalidate a cache: what could be done to of-watchdog to support these methods?

**Comment by lmoran** [comment]
I solved the issue putting NGINX in front of OpenFaas, which is arguably a good solution.
